### DISCUSSION TITLE
Prune master branch in tesseract and tesseract_planning

### DISCUSSION THREAD

**Discussion by johnwason** [tesseract-robotics/tesseract#737] [Ideas]
The master branches for tesseract and tesseract_planning are downloading almost a gigabyte every clone. It may be useful to save the current history to a side branch and prune the master branches so the downloads aren't so large. `vcs` doesn't have an option to specify a depth so setting a depth isn't practical.

**Comment by Levi-Armstrong** [comment]
What about running [this](https://git-scm.com/docs/git-maintenance)?

**Reply by johnwason** [reply]
I don't think that prevents the full history from being downloaded during each clone. I am seeing well over 1 GB of data being downloaded every clone for each repo, when the code itself is only 30 MB for tesseract and 2 MB for tesseract_planning.

**Reply by Levi-Armstrong** [reply]
I believe this will do the same thing as [git gc --aggressive](http://www.kernel.org/pub/software/scm/git/docs/git-gc.html) and then I can force push.

**Reply by johnwason** [reply]
Ok, we can see how much it reduces the download size. I would keep a backup in another branch or locally of the original.

**Reply by Levi-Armstrong** [reply]
@gavanderhoorn do you have any experience with the git clean tools?

**Reply by gavanderhoorn** [reply]
> I would keep a backup in another branch [..]

I'm not sure that would be possible: as long as there are references to objects, Git will keep them around. Only a complete prune of an object will remove it.

@Levi-Armstrong: I would probably run a `git-filter-branch` with suitable arguments.

That will rewrite history, so it's likely to break everyone's local checkout (in the sense they will see a "forced update", and references to old commits may no longer work). It depends a bit on how disconnected the branches are that are going to be changed.

**Comment by gavanderhoorn** [comment]
I believe this was already reported in #603.
